### DISCUSSION TITLE
Need help with errors during connection

### DISCUSSION THREAD

**Discussion by adevasks** [microsoft/PowerPlatform-DataverseServiceClient#356] [Q&A] [answered]
Hi,
I'm trying to connect to an environment as an application user, using a connection string and the ServiceClient class. I have a reference environment where this is working. It is not working with the environment I'm trying to use it with as well. I receive multiple errors during creating of the ServiceClient that don't tell me much about what's going wrong. Maybe someone can lead me to the right setting I have to change in the environment, or what else I have to do. The application user / secret / App registration setup should be the same in both environments.
Here are the errors I receive (unfortunately in German, but I'll try to translate them):
Microsoft.PowerPlatform.Dataverse.Client.Utils.DataverseConnectionException: 'Failed to connect to Dataverse'
1 of 5 Inner Exceptions:  AggregateException: Mindestens ein Fehler ist aufgetreten. (At least one error occured)
2: DataverseConnectionException: Exception - Fault While initializing client - RefreshInstanceDetails
3: DataverseOperationException: Exception - Failed to lookup current organization data
4: MessageSecurityException: Die HTTP-Anforderung wurde mit Clientauthentifizierungsschema "Anonymous" nicht zugelassen.
(The HTTP request was not accepted with client authentification scheme "anonymous")
5: WebException: Der Remoteserver hat einen Fehler zurückgegeben: (403) Unzulässig. (The remote serve returned an error 403).
Thank you!

**Comment by MattB-msft** [comment, marked as answer]
That error indicates that the server you're talking to is in administrative mode, or your application user is not registered in that environment.

**Reply by adevasks** [reply]
Thank you very much! Sorry for the late response, I didn't receive an email notification about your comment. You were right, the Application User did not have the role of System Administrator, so he couldn't log into an environment in administration mode. Makes sense!
For everyone else trying your solution: Give the system some time to accept the deactivation of administration mode.
